Lib/test/test_venv.py

```diff
@@ -335,6 +341,27 @@ def test_executable_symlinks(self):
             'import sys; print(sys.executable)'])
         self.assertEqual(out.strip(), envpy.encode())
 
+    @requireAlternativePythonInPath
+    def test_custom_executable(self):
+        alt_python = shutil.which("python")
+
+        # sanity check
+        self.assertNotEqual(alt_python, sys._base_executable)
+
+        builder = venv.EnvBuilder()
+        self.assertNotEqual(alt_python, builder.executable)
+        self.assertEqual(sys._base_executable, builder.executable)
+
+        builder = venv.EnvBuilder(executable=alt_python)
+        self.assertEqual(alt_python, builder.executable)
+        self.assertNotEqual(sys._base_executable, builder.executable)
+
+        # make sure it can be created with the alternative executable
+        rmtree(self.env_dir)
+        self.run_with_capture(venv.create, self.env_dir)
```

Comment:
This won't work because `venv.create()` won't call the builder you just created. Which makes me think:

1. `venv.create()` should also have the `executable` keyword argument, and pass it through to the builder that it creates.
2. The `main` function should parse an additional `--executable` argument to allow an executable to be passed in from the command line, and pass that to the builder it creates.

Also, I would suggest structuring the test as follows:

1. Remove the `requireAlternativePython` decorator and function.
2. Instead, do something along the lines of
```
    # create temp dir to hold venv
    found_a_python = False
    for minor in range(3, 11):
        executable = 'python3.%s' % minor
        alt_python = shutil.which(executable)
        if alt_python and alt_python != sys._base_executable:
            found_a_python = True
            # create builder with executable=alt_python
            # remove venv dir
            # create venv using the builder
            # check that venv creation appeared to work correctly
            # check that venv creation used the correct python interpreter by invoking it with
            # <venv_dir>/bin/python --version and checking what comes back is what's expected
    if not found_a_python:
        raise unittest.SkipTest('No alternative Python interpreters found for venv executable test')
```

Reply:
Yes, of course, will fix!

Reply:
Possibly also add a "python3" just in case there's no "python3.X" available.